RR/milestone-01-code-refactoring: removed deprecated code and refactor setting related files

rupalirajput2597 opened the pull request:
<!-- This template is to serve as a comprehensive outline for documenting updates to the code. Some of these sections may not be relevant to your PR. In that case, document it inline with `N/A`  -->

## What does this PR do?
<!-- Provide a clear and concise description of what this PR accomplishes -->
1. Removes deprecated code
2. Refactors Setting related screen with svelte 5
3. Setting screen deep linking implementation

### Related Issues & PRs
<!-- List the related issues, using either: `Closes #123` or `Fixes #123` , which will automatically move the specified issue to "In Review" when the PR is created -->
- Jonah's slack message : https://baragaun.slack.com/archives/C084NTKGT26/p1741819836233139

#### Key Decisions
<!-- Explain the technical decisions and alternative approaches considered for your new implementations and refactored solutions -->
- **Implemented Approach**:
  <!-- What did you decide on? -->
- **Alternatives Considered**:
  <!-- What else did you consider? -->
- **Rationale**:
  <!-- Why did you decide this was the most appropriate solution? -->

## Breaking Changes
<!-- Do others need to do something to successfully incorporate your changes into their work? -->
- [ ] This PR includes breaking changes
  <!-- What do they need to do? -->
No breaking changes so far

### Limitations & Concerns
<!-- Describe any limitations, potential edge cases, or situations where your solution might break -->
- **Known limitations**:
1. Test cases are missing
2. data layer Integration is pending
- **Edge cases**:
- **Potential risks**:

---

### UI Changes
<!-- If this PR modifies the UI, provide screenshots and descriptions -->
- [x] This PR includes UI changes
<!-- Include screenshots in the table below, include before & after applicable -->

### Screenshots
| Description | Screenshot |
|-------------|------------|
| Before      |  
<img width="1468" alt="Screenshot 2025-03-13 at 10 08 43 PM" src="https://github.com/user-attachments/assets/19b7ee09-5bbc-4469-b3a7-23c980b2de75" />

| After       | 
<img width="1470" alt="Screenshot 2025-03-13 at 10 25 18 PM" src="https://github.com/user-attachments/assets/2f7a2fc6-72a8-4602-a6e0-635429a832d0" />

---

### Reviewers
<!-- Tag at least 2 others for review -->
- [x] Reviewer 1: @jonahtwalker 
- [x] Reviewer 2: @shadowbrush 
- [x] Reviewer 3: @jahanvi-baragaun 
- [x] Reviewer 4: @RaghvindYadav 
- [x] Reviewer 5: @evanschimpf 
- [x] Reviewer 6: @rashmit530 
- [x] Reviewer 7: @Copilot

### Next Steps
<!-- What needs to happen after this PR is merged? -->
- [x] Follow-up tasks are created
- Just one task is created in issues there will be more integration tasks
  - [x] https://github.com/baragaun/first-spark-app/issues/43

### How to Test
<!-- Provide detailed steps for QA to test the changes -->
1. Prerequisites:
  <!-- e.g. Do I need to be a specific role? -->
4. Test steps:
   <!-- What is the quickest path to testing your solution? -->
5. Expected results:
   <!-- Is there any nuance that should be clarified? If so, can the implementation get more simple instead? -->

## Final Checklist
- [x] Linked related issues
- [x] Rebased with target branch
- [x] No lint errors
- [x] QA'd in multiple browsers
- [ ] Added/updated tests (if applicable)
- [x] Mobile responsive (if applicable)
- [x] Updated documentation (if applicable)
- [x] Notified the Team of this updates readiness

---

## AI Usage
<!-- Document any AI tools you used to develop this -->
- [x] AI tools were used in development
  - **Tool used**:
    <!-- Specify any model used -->
  VS code extension - Augment code
  - **Prompts used**:
    <!-- Format these inclusions as quotes, like: `> Provide  -->
    - **Initial prompt**: 

-     Fix the selected code. The IDE reports the following issues on L11: 'page' is deprecated.
-     I wants to show http://localhost:5173/settings/account and http://localhost:5173/settings/notification when user. selects account and notification tab from settings screen, right now we are just showing http://localhost:5173/settings url. can you please do these changes without breaking existing functionality. please do follow svelte-complete-medium.txt file before doing any changes. I want pure svelte 5 code
-     I hope you have followed the svelte-complete-medium for svelte 5 changes in setting files
-     I wan you to analyse @src/lib/components/settings-ui/notification-settings.svelte file as per svelte-complete-medium.txt and remember do not remove any commented code from the file as I have kept it for future use
-     I wan you to analyse @src/lib/components/settings-ui/account-settings.svelte file as per svelte-complete-medium.txt. and update if any non svelte 5 code and remember do not remove any commented code from the file I have kept it for future use
-     dialog-content.svelte file as per svelte-complete-medium.txt. and update if any non svelte 5 code
-     dialog.svelte file as per svelte-complete-medium.txt. and update if any non svelte 5 code
-     some more to update remaining dialog files

- [ ] No AI tools were used

## References
<!-- List any helpful references used during development -->
- **Documentation**:
- **Discussions**:
- **Related PRs**:

Commit:
RR/milestone-01-code-refactoring: removed deprecated code and refactore milestone-01 files

jonahtwalker commented on src/routes/faq/+page.svelte:
```diff
@@ -6,21 +6,25 @@
-      console.log(`FAQ sections open: ${getOpenQuestionsCount()}`);
+  $effect.root(() => {
+    if (typeof window !== 'undefined' && openQuestionsCount > 0) {
+      console.log(`FAQ sections open: ${openQuestionsCount}`);
```
Do we need this log?

rupalirajput2597 replied:
It is not needed. will remove it!

jonahtwalker commented on src/routes/settings/+layout.svelte:
```diff
@@ -0,0 +1,52 @@
+  import { getStores } from '$app/stores';
+  import { goto } from '$app/navigation';
+
+  const { page } = getStores();
```
Why aren't you using `state`?

rupalirajput2597 replied:
I initially asked the AI to remove deprecated changes, but I didn't realize it wasn't Svelte 5 code. After reviewing the [migration guide for app-store](https://svelte.dev/docs/kit/migrating-to-sveltekit-2#SvelteKit-2.12:-$app-stores-deprecated), I now understand the necessary updates and will make the corrections accordingly. Thanks for catching that, Jonah!

jonahtwalker commented on src/lib/components/app-sidebar.svelte:
```diff
@@ -8,56 +8,41 @@
   import { fly } from 'svelte/transition';
   import { quartOut } from 'svelte/easing';
-  import { page } from '$app/stores';
+  import { getStores } from '$app/stores';
```
Why do you use Stores instead of State?

rupalirajput2597 replied:
Same here, will make the corrections according to [migration guide for app-store](https://svelte.dev/docs/kit/migrating-to-sveltekit-2#SvelteKit-2.12:-$app-stores-deprecated). Thank you!

jonahtwalker commented on src/lib/components/app-sidebar.svelte:
```diff
@@ -8,56 +8,41 @@
+  let currentPath = $derived.by(() => $page.url.pathname);
+  let sidebarState = $derived.by(() => sidebar.state);
+  let itemsList = $derived.by(() => Object.values(items));
+  let isActive = (url: string) => {
```
The component from shad already handles this, see [this line in my pr](https://github.com/baragaun/first-spark-app/pull/44/files#:~:text=isActive%3D%7BactiveItem.title%20%3D%3D%20item.title%7D)

rupalirajput2597 replied:
ohh, I got it. Let me update my code the same way

jonahtwalker commented on src/svelte-coding-instructions.txt:
Is this being removed in favor of the medium Svelte docs file?

rupalirajput2597 replied:
I have removed this, because It is already there in `docs` folder, so just removed extra one

jonahtwalker reviewed (changes requested):
Thanks Rupali. Did you look at [the component docs](https://next.shadcn-svelte.com/docs/components/dialog) before making these updates?

If we've added Svelte4 components from shad via `shadcn-svelte@latest` instead of using the Svelte5 versions via `shadcn-svelte@next`, we should just replace the components entirely, and not further customize them ourselves. Take a look at [`dialog-description.svelte`](https://github.com/huntabyte/shadcn-svelte/blob/main/sites/docs/src/lib/registry/default/ui/dialog/dialog-description.svelte), for instance.

I am going to QA and I'll get back with additional feedback. Thanks again!

Commit:
review comment changes

Commit:
installed and used shadcn dialog

rupalirajput2597 commented:
> Thanks Rupali. Did you look at [the component docs](https://next.shadcn-svelte.com/docs/components/dialog) before making these updates?
> 
> If we've added Svelte4 components from shad via `shadcn-svelte@latest` instead of using the Svelte5 versions via `shadcn-svelte@next`, we should just replace the components entirely, and not further customize them ourselves. Take a look at [`dialog-description.svelte`](https://github.com/huntabyte/shadcn-svelte/blob/main/sites/docs/src/lib/registry/default/ui/dialog/dialog-description.svelte), for instance.
> 
> I am going to QA and I'll get back with additional feedback. Thanks again!

Hello Jonah,

Thank you for reviewing my PR and providing valuable feedback. I have updated the Dialog component to the Svelte 5 version using shadcn-svelte@next and implemented the other suggested changes as well. I would appreciate it if you could review it again and let me know if you have any further feedback.

Looking forward to your thoughts!

jonahtwalker commented:
Thank you Rupali! I've cherry picked some things here and incorporated them into my PR, #44 

Will you give that a review when you've some time? I've left comments on just about all of the files. I'd like to do a bit more in the menus and then merge that in as soon as we can so we can get to the node-client stuff this week.

Thanks again! 👍🏼

jonahtwalker commented:
Updates from this PR have been combined with #44
